finagle-core/src/main/scala/com/twitter/finagle/Name.scala

```diff
@@ -39,7 +39,11 @@ object Name {
    *
    * Equality of two Names is delegated to `id`. Two Bound instances
    * are equal whenever their `id`s are. `id` identifies the `addr`
-   * and not the `path`.
+   * and not the `path`.  If the `id` is a [[com.twitter.finagle.Name.Path
+   * Path]], it should only contain *bound*--not residual--path components.
```

Comment:
er, to be more precise, I don't know what a residual path is.  Can we add a description of what a residual path is, and what it might be used for?